### PR TITLE
basic genserver structure

### PR DESCRIPTION
**Pull request by craigp**

### CODE REVIEW

File: snippets/elixir.snippets [hunk 1]
```diff
@@ -197,3 +197,15 @@ snippet >i pipe to inspect
 	|> IO.inspect
 snippet >il pipe to inspect with label
 	|> IO.inspect(label: "${1:label}")
+snippet gs basic genserver structure
```
Comment (lpil) [inline snippets/elixir.snippets]: This is going to be used infrequently, so could we give it a longer trigger please? :)

File: snippets/elixir.snippets [hunk 2]
```diff
@@ -197,3 +197,15 @@ snippet >i pipe to inspect
...
+	use GenServer
+
+	@doc false
+	def start_link do
+		GenServer.start_link(__MODULE__, ${1:Map.new}, name: __MODULE__)
```
Comment (lpil) [inline snippets/elixir.snippets]: We should not default to naming the process. Naming process by default removes the ability to test them concurrently.

File: snippets/elixir.snippets (hunk 1 above)
Comment (craigp) [reply snippets/elixir.snippets]: Sure - in my personal snippets it was `genserver`, so I'll change it to that

File: snippets/elixir.snippets (hunk 2 above)
Comment (craigp) [reply snippets/elixir.snippets]: True, I just don't often create unnamed processes - I'll change it.

**Pull request merged**